compiler/ilgen/TypeDictionary.hpp

```diff
@@ -75,6 +75,53 @@ class IlType
    };
 
 
+/**
+ * @brief Convenience API for defining JitBuilder structs from C/C++ structs (PODs)
+ * 
+ * These macros simply allow the name of C/C++ structs and fields to be used to
+ * define JitBuilder structs. This can help ensure a consistent API between a
+ * VM struct and JitBuilder's representation of the same struct. Their definitions
+ * expand to calls to `TR::TypeDictionary` methods that create a representation
+ * corresponding to that specified type.
+ * 
+ * ## Usage
+ * 
+ * Given the following struct:
+ * 
+ * ```c++
+ * struct MyStruct {
+ *    int32_t id;
+ *    int32_t* val;
+ * };
+ * 
+ * a JitBuilder representation of this struct can be defined as follows:
+ * 
+ * ```c++
+ * TR::TypeDictionary d;
+ * d.DEFINE_STRUCT(MyStruct);
```

Comment:
The intention of the macros is to provide a more convenient interface for defining JitBuilder macros from C macros. They are not undefined at the end of the header as they are intended to consumed by application code (see LinkedList example). It's true that `DEFINE_STRUCT` provides relatively little value, but its presence makes the API consistent. Also, the definition of `DEFINE_FIELD` may (eventually) be changed so that the type of a field need not be specified by the user.